docs/rep/node-agent/README.md

```diff
@@ -0,0 +1,99 @@
+# RIK - Node Agent Architecture (Riklet)
+
+This document provides a technical reflection about our component, the **node agent**.
+
+Our component will be installed on each node of a cluster and it will be composed of 4 sub-components :
+
+- **Node Manager**
+- **Node Runtime Memory**
+- **Node Proxy**
+- **Node Monitor**
+
+We will describe these components in more details throughout this document. Here is a diagram to illustrate how these components works together :
+
+![High-level view of Rikelet](assets/node-schema.png)
+
+## 1) Node Manager
+
+The Node Manager is the main sub-component of the Riklet. It should be able to :
+
+- Interprete Scheduler instructions
+- Run / Stop an application
+- List pods in the node
+
+## 2) Node Runtime Memory
+
+![](assets/nrm.png)
+
+The Node Runtime Memory is a simple runtime memory which allow us to manage all the running pods on the node. This component allow the Node Manager to get Pods instance fastly to execute actions on them. It saves the state of each pods.
+
+## 3) Node Monitor
+
+The node monitor is responsible to aggregate and return node / pods metrics. It will be useful for the scheduler to get system resources informations about the node.
+
```

Comment:
Useful is not a strong enough word here :-) The scheduler won't be able to make accurate decisions without that information from each node.

Some comments:

- There are node metrics and there are pod metrics. The scheduler only cares about node metrics, while the controller only cares about pod metrics/status. Who are we going to report those payloads described below to? The controller or the scheduler?
- Those resources/metrics/stats are dynamic and change as pods come and go. There are 2 models for reporting them:
   - A push model where the node asynchronously pushes the data, only when it changes, to the scheduler and the controller
   - A pull model where the controller and scheduler regularly request for status updates

Again,  my suggestions:

- Have the node specific metrics reported to the scheduler
- Have the pod specific metrics reported to the controller
- Use a push model where the node decides when and what to report to the controller and the scheduler. This guarantees a much more accurate view of the cluster from the controller and the scheduler, less network traffic and a more flexible way of adding and removing nodes from a cluster.